Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -695,23 +714,25 @@ def self.latest_version(
 
         next if strategy.blank?
 
-        homebrew_curl = case strategy_name
-        when "PageMatch", "HeaderMatch"
-          use_homebrew_curl?(referenced_package, url)
+        if (livecheck_homebrew_curl = livecheck_options.homebrew_curl).nil?
+          case strategy_name
+          when "PageMatch", "HeaderMatch"
+            if (homebrew_curl = use_homebrew_curl?(referenced_package, url))
+              livecheck_options = livecheck_options.merge({ homebrew_curl: })
+              livecheck_homebrew_curl = homebrew_curl
+            end
+          end
         end
-        puts "Homebrew curl?:   Yes" if debug && homebrew_curl.present?
-
-        strategy_args = {
-          regex:         livecheck_regex,
-          url_options:   livecheck_url_options,
-          homebrew_curl:,
-        }
-        # TODO: Set `cask`/`url` args based on the presence of the keyword arg
-        # in the strategy's `#find_versions` method once we figure out why
-        # `strategy.method(:find_versions).parameters` isn't working as
-        # expected.
-        strategy_args[:cask] = cask if strategy_name == "ExtractPlist" && cask.present?
-        strategy_args[:url] = url
+        puts "Homebrew curl?:   #{livecheck_homebrew_curl ? "Yes" : "No"}" if debug && !livecheck_homebrew_curl.nil?
+
+        # Only use arguments that the strategy's `#find_versions` method
+        # supports
+        find_versions_parameters = T.must(livecheck_find_versions_parameters[strategy])
```

Comment:
```suggestion
        find_versions_parameters = livecheck_find_versions_parameters.fetch(strategy)
```
…though see comments above about alternatives to this approach